tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -831,56 +875,33 @@ def generator_py_func(iterator_id):
         """A `py_func` that will be called to invoke the iterator."""
         # `next()` raises `StopIteration` when there are no more
         # elements remaining to be generated.
-        values = next(generator_state.get_iterator(iterator_id))
+        values = next(generator_state.get_iterator(iterator_id.numpy()))
 
-        # Use the same _convert function from the py_func() implementation to
-        # convert the returned values to arrays early, so that we can inspect
-        # their values.
         try:
-          flattened_values = nest.flatten_up_to(output_types, values)
+          values = structure.normalize_element(values, output_signature)
```

Comment:
Previously, `from_generator` would convert generated elements to match the types specified with `output_types`, but with this change `from_generator` will require an exact dtype match. The docs say 

> The elements generated by generator must be compatible with the given output_types

https://www.tensorflow.org/api_docs/python/tf/data/Dataset#from_generator

Can we preserve the original behavior of converting compatible dtypes, and add a test to verify that it works?

Reply:
@aaudiber Can you give me the example (that we can later convert to the test) of this working in old version and now working in the new one?

Reply:
I looked into it further and I think the current code is still doing implicit conversions between compatible types, so no change is needed. Sorry for the confusion.

Reply:
no worries